### PR TITLE
Implement interface SnapshotPvtdataHashesConsumer for pvtdata store

### PR DESCRIPTION
**Pull request by manish-sethi**
#### Type of change
- New feature

#### Description
This PR prepares the ground for populating private data store from snapshot files.

#### Additional details
In the followup PR(s), 
- The hashed data snapshot importer (introduced in this PR) will be integrated with the path where the hashed data is loaded in to statedb
- The reconciler will be integrated to work with the hashed data populated into the store (instead of from the blockstore) for the data below the snapshot height

### CODE REVIEW

File: core/ledger/pvtdatastorage/store.go [hunk 1]
```diff
@@ -577,6 +584,10 @@ func (s *Store) purgeExpiredData(minBlkNum, maxBlkNum uint64) error {
 			)
 		}
 
+		for _, bootKVHashesKey := range bootKVHashesKeys {
+			batch.Delete(encodeBootKVHashesKey(bootKVHashesKey))
+		}
```
Comment (cendhu) [inline core/ledger/pvtdatastorage/store.go:589]: good to add a test for this.

File: core/ledger/pvtdatastorage/snapshot_data_importer.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+			)
+	}
+
+	if dataBlockNum <= history[len(history)-1].configBlockNum {
+		return false,
+			errors.Errorf(
+				"unexpected error - no collection config found below block number [%d] for <namespace=%s, collection=%s>",
+				dataBlockNum, namespace, collection,
+			)
+	}
```
Comment (cendhu) [inline core/ledger/pvtdatastorage/snapshot_data_importer.go:205]: Are we storing all eligibility history to tackle this error case?

File: core/ledger/pvtdatastorage/snapshot_data_importer_test.go [hunk 3]
```diff
@@ -0,0 +1,1034 @@
...
+
+	t.Run("ineligible-then-eligible", func(t *testing.T) {
+		eligibilityBTLCache.eligibilityHistory = map[nsColl][]*eligibility{
+			{ns: "ns", coll: "coll"}: {
+				{configBlockNum: 10, isEligible: true},
+				{configBlockNum: 5, isEligible: false},
+			},
+		}
+
+		for i := 6; i < 15; i++ {
```
Comment (cendhu) [inline core/ledger/pvtdatastorage/snapshot_data_importer_test.go:633]: even for 1 to 5, this test would pass, right?

**Review by cendhu — COMMENTED** [review]
Thanks, Manish. A few minor nits and some clarification questions.

File: core/ledger/pvtdatastorage/snapshot_data_importer_test.go (hunk 3 above)
Comment (manish-sethi) [reply core/ledger/pvtdatastorage/snapshot_data_importer_test.go:633]: No, it should expect an error for that range - `unexpected error - no collection config found below block number ...`

**Review by cendhu — APPROVED** [review]
LGTM. Thanks, Manish.

**Pull request merged**